rustls/src/common_state.rs

```diff
@@ -257,14 +257,19 @@ impl CommonState {
         };
         match state.handle(&mut cx, msg) {
             Ok(next) => Ok(next),
-            Err(e @ Error::InappropriateMessage { .. })
-            | Err(e @ Error::InappropriateHandshakeMessage { .. }) => {
-                Err(self.send_fatal_alert(AlertDescription::UnexpectedMessage, e))
-            }
             Err(e) => Err(e),
         }
     }
 
+    pub(crate) fn send_fatal_alert_for_error(&mut self, error: &Error) {
```

Comment:
So this silently does nothing when the `TryFrom` impl yields `Err(())`. Maybe the name should be adapted to reflect that? Like `maybe_send_fatal_alert()`? (The `for_error` stanza doesn't seem super-useful given that it's part of the signature, but I assume it was needed to disambiguate from existing methods.)